display_list/display_list_rtree.cc

```diff
@@ -8,43 +8,166 @@
 
 namespace flutter {
 
-DlRTree::DlRTree() : bbh_{SkRTreeFactory{}()}, all_ops_count_(0) {}
+DlRTree::DlRTree(const SkRect rects[],
+                 int N,
+                 const int ids[],
+                 bool p(int),
+                 int invalid_id)
+    : leaf_count_(0), invalid_id_(invalid_id) {
+  if (N <= 0) {
```

Comment:
I've determined empirically that 1952257857 is the largest number of leaves we can deal with without the number of nodes overflowing an int. I could switch this all to uint32 or uint64 internally or simply reject the construction of an R-Tree with more leaves than that.

Thoughts?

Reply:
It looks like 2^31 leaves generates less than 2^32 total nodes due to the reduction by 11 on every generation of nodes, so using uint32_t internally might fix this overflow potential.

Reply:
fixed using `uint32_t` in latest commit for node indexing past the leaf counting stage...